### ISSUE TITLE
Viktor_Cortess - function lockQuote increments nonce incorrectly

### ISSUE THREAD

**Issue by sherlock-admin** [sherlock-audit/2023-06-symmetrical-judging#27]
Viktor_Cortess

medium

# function lockQuote increments nonce incorrectly

## Summary

Mapping partyBNonces uses address 0 as a PartyB address incrementing nonce

## Vulnerability Detail

After the execution of sendQuote() function from PartyAFacetImpl library newly created quote has address(0) as a PartyB:

https://github.com/sherlock-audit/2023-06-symmetrical/blob/main/symmio-core/contracts/facets/PartyA/PartyAFacetImpl.sol#L105

When PartyB locks this quote it calls lockQuote() function:

https://github.com/sherlock-audit/2023-06-symmetrical/blob/main/symmio-core/contracts/facets/PartyB/PartyBFacetImpl.sol#L22-L38

As we can see it uses msg.sender as PartyB in line 27:

    27: LibMuon.verifyPartyBUpnl(upnlSig, msg.sender, quote.partyA);

But in the if statement it uses quote.PartyB i.e. address(0):

    30: accountLayout.partyBNonces[quote.partyB][quote.partyA] += 1;

And only after that, it assigns the value of msg.sender to the quote.partyB

    34: quote.partyB = msg.sender;

## Impact

Verifying functions from Muon Library will work incorrectly.

## Tool used

Manual Review

## Recommendation

Assign the value of msg.sender to the quote.partyB before incrementing nonce or use a msg.sender as a quote.PartyB incrementing nonce.

    quote.partyB = msg.sender;
    if (increaseNonce) { 
         accountLayout.partyBNonces[quote.partyB][quote.partyA] += 1; 
     } 
or

    if (increaseNonce) { 
         accountLayout.partyBNonces[msg.sender][quote.partyA] += 1; 
     } 

Duplicate of #123
